test_runner/src/main/kotlin/ftl/run/NewTestRun.kt

```diff
@@ -15,18 +15,17 @@ import ftl.util.FlankTimeoutError
 import kotlinx.coroutines.TimeoutCancellationException
 import kotlinx.coroutines.withTimeoutOrNull
 
-suspend fun newTestRun(args: IArgs) {
-    withTimeoutOrNull(args.parsedTimeout) {
-        println(args)
-        val (matrixMap, testShardChunks, ignoredTests) = cancelTestsOnTimeout(args.project) { runTests(args) }
+suspend fun newTestRun(args: IArgs) = withTimeoutOrNull(args.parsedTimeout) {
+    println(args)
+    val (matrixMap, testShardChunks, ignoredTests) = cancelTestsOnTimeout(args.project) { runTests(args) }
 
-        if (!args.async) {
-            cancelTestsOnTimeout(args.project, matrixMap.map) { pollMatrices(matrixMap.map.keys, args).update(matrixMap) }
-            cancelTestsOnTimeout(args.project, matrixMap.map) { fetchArtifacts(matrixMap, args) }
+    if (!args.async) {
+        cancelTestsOnTimeout(args.project, matrixMap.map) { pollMatrices(matrixMap.map.keys, args).update(matrixMap) }
+        cancelTestsOnTimeout(args.project, matrixMap.map) { fetchArtifacts(matrixMap, args) }
 
-            ReportManager.generate(matrixMap, args, testShardChunks, ignoredTests)
-        }
+        ReportManager.generate(matrixMap, args, testShardChunks, ignoredTests)
     }
+    return@withTimeoutOrNull this
```

Comment:
Is there any reason for returning CoroutineScope?

Reply:
No, this is redundant I should remove it. Thanks!